torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -5737,24 +5737,33 @@ Tensor lu_unpack_backward(
 
 Tensor cat_jvp(at::ITensorListRef tensors, int64_t dim) {
   Tensor out_fw_grad;
+  at::TensorOptions options;
 
   auto materialized = tensors.materialize();
   auto any_defined = false;
   for (const Tensor& t : materialized) {
-    any_defined |= isFwGradDefined(t);
+    if (isFwGradDefined(t)) {
+      any_defined = true;
+      options = t.options();
+      break;
+    }
   }
 
   if (any_defined) {
     std::vector<Tensor> fw_grads;
 
     for (const Tensor& t : materialized) {
+      if (t.sizes() == IntArrayRef({0})) {
+        continue;
+      }
       fw_grads.push_back(
           isFwGradDefined(t)
               ? t._fw_grad(/*level*/ 0)
               : at::_efficientzerotensor(t.sizes(), t.options()));
     }
 
-    out_fw_grad = at::cat(fw_grads, dim);
+    out_fw_grad = fw_grads.size() > 0 ? at::cat(fw_grads, dim)
+                                      : at::_efficientzerotensor({0}, options);
```

Comment:
What is the behavior here?
- we always take the tensor options of the last tensor?
- what happens for forward normally, we should probably mirror that behavior there

Reply:
1. The first forward grad defined tensor (please see above).
2. The logic in forward is a bit more complex but basically similar:

https://github.com/pytorch/pytorch/blob/7a8d0ccddff9480b3ca86bfed2f77c5ad43f35b2/aten/src/ATen/native/TensorShape.cpp#L258-L265

where `cat_should_skip_tensor` checks whether the tensor is empty.

Reply:
The difference is that the forward uses the first non-empty tensor's option, while the jvp uses the first tensor with forward grad defined (certainly empty in this very situation).